src/log-store/src/raft_engine/backend.rs

```diff
@@ -43,11 +43,36 @@ pub struct RaftEngineBackend {
 
 impl RaftEngineBackend {
     pub fn try_open_with_cfg(config: Config) -> error::Result<Self> {
+        Self::check_config(&config)?;
+
         let engine = Engine::open(config).context(RaftEngineSnafu)?;
         Ok(Self {
             engine: RwLock::new(engine),
         })
     }
+
+    fn check_config(config: &Config) -> error::Result<()> {
```

Comment:
This is somewhat of a monkey patch, but the `std::io::Error` generated from ERRNO doesn't have the path info. RaftEngine's code is complex to enrich the info, as well as its IO error variant doesn't have a message field. Let's fix this case specially, although I may prefer a global utility for enriching IO errors.

Reply:
Most of fs operations are through OpenDAL which has a better error behavior. But unfortunately raft-engine is an exception